src/jdk.jfr/share/classes/jdk/jfr/internal/RepositoryChunk.java

```diff
@@ -174,4 +180,25 @@ public long getCurrentFileSize() {
             return 0L;
         }
     }
+
+    boolean isMissingFile() {
+        try {
+            if (!SecuritySupport.exists(chunkFile)) {
+                return true;
+            }
+        } catch (IOException ioe) {
+            // ignore
+        }
+        return false;
+    }
+
+    void emitMissingChunkFileEvent() {
+        emitMissingChunkFileEvent(this.endTime);
+    }
+
+    void emitMissingChunkFileEvent(Instant endTime) {
+        String message = "Chunkfile \""+ chunkFile.toString() + "\" is missing.";
+        Logger.log(LogTag.JFR, LogLevel.ERROR, message + " Data loss might occur from " + startTime.toString() + (endTime != null ? " to " + endTime.toString() : ""));
+        ErrorThrownEvent.commit(startTime.getNano(), endTime != null ? Duration.between(startTime, endTime).toNanos() : 0L, message, MissingChunkFileError.class);
```

Comment:
I've skipped the MissingChunkFileError event when finding missing files during dumping (it would've only shown up in subsequent recordings anyways, so of little use). I instead throw the error when finding missing files in-flight; I think it's important to have something that hints to user that a recording might be missing data, and the error will show up in the recording that might be missing some data.